lively.merger/merger.js

```diff
@@ -1,107 +1,121 @@
-  ) {
-    const morphA = $world.submorphs.filter(morph => morph.id === morphAid)[0];
-    if (!morphA) {
-      throw new Error(`Cannot merge morphs, morphA with id ${morphAid} not found`);
-    }
-    const morphB = $world.submorphs.filter(morph => morph.id === morphBid)[0];
-    if (!morphB) {
-      throw new Error(`Cannot merge morphs, morphB with id ${morphBid} not found`);
-    }
-    return this.mergeMorphs(
-      morphA, 
-      morphB, 
-      (properties, mergeConflicts) => onMergeResult(properties, mergeConflicts));
+  if (JSON.stringify(morphA.styleClasses) !== JSON.stringify(morphB.styleClasses)) {
+    throw new Error('Cannot merge morphs, styleclasses differ');
   }
 
-  static mergeMorphsIntoA (morphA, morphB) {
-    return this.mergeMorphs(morphA, morphB, (properties, mergeConflicts) => {
-      Object.keys(properties).forEach(key => {
-        morphA[key] = properties[key];
-      });
-      return morphA;
-    });
-  }
+  let parentMorph = await getLowestCommonAncestor(morphA, morphB);
 
-  static mergeMorphsIntoB (morphA, morphB) {
-    return this.mergeMorphsIntoA(morphB, morphA);
-  }
+  let propertiesmorphA = propertiesFromMorph(morphA);
+  let propertiesmorphB = propertiesFromMorph(morphB);
+  let propertiesParentMorph = propertiesFromMorph(parentMorph);
+
+  let result = merge(
+    propertiesParentMorph,
+    propertiesmorphA,
+    propertiesmorphB);
+    // TODO conflict resolve
+  return onMergeResult(result.properties, result.mergeConflicts);
+}
 
-  static mergeMorphs (
+export async function mergeMorphsWithIds (
+  morphAid, 
+  morphBid, 
+  onMergeResult = (properties, mergeConficts) => new Morph(properties)
+) {
+  const morphA = $world.submorphs.filter(morph => morph.id === morphAid)[0];
+  if (!morphA) {
+    throw new Error(`Cannot merge morphs, morphA with id ${morphAid} not found`);
+  }
+  const morphB = $world.submorphs.filter(morph => morph.id === morphBid)[0];
+  if (!morphB) {
+    throw new Error(`Cannot merge morphs, morphB with id ${morphBid} not found`);
+  }
+  return mergeMorphs(
     morphA, 
     morphB, 
-    onMergeResult = (properties, mergeConficts) => new Morph(properties)
-  ) {
-    if (!morphA.isMorph || !morphB.isMorph) {
-      throw new Error('Cannot merge objects that are not morphs');
-    }
-    if (JSON.stringify(morphA.styleClasses) !== JSON.stringify(morphB.styleClasses)) {
-      throw new Error('Cannot merge morphs, styleclasses differ');
-    }
+    (properties, mergeConflicts) => onMergeResult(properties, mergeConflicts)).catch(error => { throw error; });
+}
 
-    let parentMorph = this.getLowestCommonAncestor(morphA, morphB);
+export async function mergeMorphsIntoA (morphA, morphB) {
+  return mergeMorphs(morphA, morphB, (properties, mergeConflicts) => {
+    Object.keys(properties).forEach(key => {
+      morphA[key] = properties[key];
+    });
+    return morphA;
+  }).catch(error => console.log(error));
```

Comment:
Please remove any console.log

```suggestion
  });
```